src/utils/formUtil.ts

```diff
@@ -680,7 +680,7 @@ export function getComponent(
 ): (Component | undefined) {
   let result;
   eachComponent(components, (component: Component, path: any) => {
-    if ((path === key) || (component.path === key)) {
+    if ((path === key) || (component.path === key) || (component.key === key)) {
```

Comment:
@AlexeyNikipelau I found that this logic doesn't now seem to find nested components properly, since the 'path' is an absolute one. Do you feel fine adding a check for matching component key?

Reply:
I believe this is going to be fine. This is for the function "getComponent" which has always been a "fuzzy" search for the component. The only time this may be a problem is if you have two components with the same "key" which is only possible if you add a 'non-data' nested component (like a panel). Maybe we should add the following check for "input"

```js
(component.input && (component.key === key))
```